File: invoice/src/main/java/com/aact/caps/service/CodeService.java
package com.aact.caps.service;

import com.aact.caps.dto.CodeDto;
import com.aact.caps.entity.Code;
import com.aact.caps.repository.jpa.CodeRepository;
import org.springframework.beans.BeanUtils;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.stream.Collectors;

/**
 * 공통 코드 서비스
 */
@Service
@Transactional(readOnly = true)
public class CodeService {

    @Autowired
    private CodeRepository codeRepository;

    /**
     * 코드 타입별 코드 목록 조회 (사용 중인 것만)
     */
    public List<CodeDto> getCodesByType(String codeType) {
        List<Code> codes = codeRepository.findByCodeTypeAndUseYn(codeType);
        return codes.stream()
                .map(this::convertToDto)
                .collect(Collectors.toList());
    }

    /**
     * 터미널 코드 목록 조회
     */
    public List<CodeDto> getTerminals() {
        return getCodesByType("TRMCD");
    }

    /**
     * 부서 코드 목록 조회
     */
    public List<CodeDto> getDepartments() {
        return getCodesByType("DPTCD");
    }

    /**
     * 직급 코드 목록 조회
     */
    public List<CodeDto> getPositions() {
        return getCodesByType("POSTK");
    }

    /**
     * CAPS 사용자 목록 조회
     */
    public List<CodeDto> getCAPSList() {
        return getCodesByType("CAPS");
    }

    /**
     * Entity를 DTO로 변환
     */
    private CodeDto convertToDto(Code code) {
        CodeDto dto = new CodeDto();
        BeanUtils.copyProperties(code, dto);
        return dto;
    }
}

File: invoice_front/src/caps/services/codeService.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8081/api/caps';

/**
 * 공통 코드 API 서비스
 */
const codeService = {
  /**
   * 터미널 코드 목록 조회 (TRMCD)
   * @returns {Promise<Array>} 터미널 코드 목록
   */
  async getTerminals() {
    try {
      const response = await axios.get(`${API_BASE_URL}/codes/terminals`);
      return response.data;
    } catch (error) {
      console.error('터미널 코드 조회 실패:', error);
      throw error;
    }
  },

  // 부서/직급 코드는 사용자 조회 시 JOIN으로 가져오므로 제거됨
  // getDepartments(), getPositions(), getCAPSList() 메서드 제거

  /**
   * 공통 코드 조회 (범용)
   * @param {string} codeType - 코드 타입 (예: TRMCD, DPTCD, POSTK, CAPS)
   * @returns {Promise<Array>} 코드 목록
   */
  async getCodesByType(codeType) {
    try {
      const response = await axios.get(`${API_BASE_URL}/codes?type=${codeType}`);
      return response.data;
    } catch (error) {
      console.error(`${codeType} 코드 조회 실패:`, error);
      throw error;
    }
  }
};

export default codeService;
